### PR TITLE
Reinforcement Learning Template

### PR DESCRIPTION
**Pull request by theory-in-progress**
- Added 2 algorithms for Reinforcement Learning: REINFORCE and Advantage Actor Critic (A2C)
- The Algorithms are implemented for the OpenAI Gym env: CarRacing-v2

### CODE REVIEW

File: src/templates/template-reinforcement-learning/advantage_actor_critic_a2c.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    import gymnasium as gym
+except ImportError:
+    raise ModuleNotFoundError("Please install opengym: pip install gymnasium[box2d]")
+
+SavedAction = namedtuple("SavedAction", ["log_prob", "value"])
+
+eps = np.finfo(np.float32).eps.item()
+
+
+class ActorCriticNetwork(nn.Module):
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply src/templates/template-reinforcement-learning/advantage_actor_critic_a2c.py:35]: yes, good point, I was also once thinking about [torchrl](https://github.com/pytorch/rl) and that we could use it. Thanks!
